### DISCUSSION TITLE
[21. 11. 09] 리포트, 역량 회의

### DISCUSSION THREAD

**Discussion by sunhpark42** [woowacourse/prolog#599] [Feature]
## 리포트

- 오버뷰에는 역량 그래프와 학습로그 목록
- 리포트 탭에서는 가장 최근의 리포트를 보여줌. 리포트 목록 페이지는 혼선이 있을 것 같아 삭제하는 게 좋을 것 같다.
- 이력, 기록이 남는 의미상 이력 남기기 시 날짜가 prefix되어야 할 것 같음.
- 역량이 수정됐을 때 기존 것이 변경되면 안됨.
- 역량은 현재 리포트에서 관리 되고있는 역량.
- 이력으로 남은 리포트는 고정 됨. 
- 오버뷰에서 바로 수정 가능 => 수정시에 단순 수정 / 이력 남기기 선택 가능
- 리포트 목록 클릭시 가장 최근의 리포트가 최 상단
- 현재 목록은 타임라인으로 변경 (세로 UI)
- 가장 최근이 가장 위에
- 과거이력은 삭제 가능
- 과거 이력 조회시에는 수정은 없고, 삭제만 있음. 
- 새 리포트 등록 > 이력 추가 버튼 변경.
- 이력 추가 > 가장 최근의 리포트를 수정 하는 것과 같음.
- 수정을 했을 때 학습로그 목록은 0개에서 시작해야 한다고 생각함.
- 오버뷰에 있는 태그, 캘린더는?
	- 학습로그 목록으로 옮겨도 될 것 같음.
	- 더보기로 확장해서 볼 수 있게 하기.
- 최신 리포트에 리포트 작성하기 > 리포트 수정하기 버튼으로 변경

### 일정
- 페이지 별 필요한 API 목록 정리 & 전달 필요
- 정해진 마감 기한 별도로 없어서, 개인 일정 먼저 생각하기.
- API 회의 잡자~!~!~!~!~!

**Comment by gracefulBrown** [comment]
# 오버뷰 페이지 - 서니
- 최근 리포트의 역량 그래프와 학습로그 목록을 노출
- [이력 남기기] 버튼으로 각 페이지로 이동

# 학습로그 페이지 - 서니
- 오버뷰에 있던 태그, 캘린더 -> 학습로그 페이지로 이동
- 더보기로 확장시 태그 목록과 캘린더 노출

# 리포트 목록 페이지 - 티케
- 리포트 기록은 타임라인으로 변경 (세로 UI)
- 목록 상단에 [이력 남기기] 버튼이 있음

# 리포트 이력 남기기 페이지 - 티케
- 가장 최근의 리포트를 기반으로 새로운 리포트 작성하기
- 최근 리포트 가져오기 기능
  - 현재 역량이랑 최근 리포트 역량이랑 같은지 비교해서 같으면 불러와지기
  - 학습로그도 가져오기
  - 다르면 다르다고 메시지 띄우기
- 이력, 기록이 남는 의미상 이력 남기기 시 날짜가 prefix되어야 할 것 같음.

# 리포트 조회 페이지 - 티케
- 과거 이력 조회시에는 수정은 없고, 삭제만 있음. 
- 역량페이지에서 역량이 수정됐을 때 기존 것이 변경되면 안됨.
- 새 리포트 등록 버튼 제거

# 역량 페이지 - 서니
- 역량은 현재 리포트에서 관리 되고있는 역량.

**Comment by gracefulBrown** [comment]
## 백엔드 
- 리포트 저장 시 역량 안 바뀌게 하는거
- 최근 리포트 가져오기 API 추가

**Comment by devhyun637** [comment]
잘 이해가 안가는 부분 앞에는 🤚 표시를 해두었습니다!

0. 사용자의 A가 가입한다. 역량을 가지고 있지 않은 상태이다 
=> 리포트 페이지 : 🤚`역량 및 학습로그를 작성해야합니다.`를 띄워주는 방향이 필요하지 않을 까 합니다.

1. 사용자 A는 역량 1, 2, 3, 4을 설정한다, 사용자 A는 학습로그 A, B, C를 작성한다.
=> 리포트 페이지 : `이력 남기기 버튼`이 보인다

2. 사용자는 리포트 페이지에 접속한다.
=> 리포트 페이지 : `이력 남기기 버튼`이 보인다.

3. 사용자 A는 현재 역량 중 보여줄 역량을 선택하고, 학습로그 목록에 등록하고 싶은 학습로그를 등록한다.

4. 사용자 A가 리포트를 등록한다.
=> 리포트 페이지 : 타임라인 첫번째 리포트가 등록된다.

5. 사용자 A는 리포트 페이지에서 기존에 등록한 리포트를 선택했을 때, 수정 및 삭제가 가능하다

6. 사용자 A는 리포트페이지에 왔을 때, `이력 남기기 버튼`을 볼 수 있다.

7. 사용자 A의 달라진 역량과 학습로그가 없으므로 `이력 남기기 버튼`은 비활성화 상태이다.

8. 사용자는 기존의 역량 1을 5로 수정한다.

🤚사용자의 이력 남기기가 가능한 상태일까요?
🤚4번에서 에서 작성한 리포트를 수정한다면, '등록된 역량 현재 역량과 다릅니다.' 라는 안내가 생기는 게 맞을까요?

**Reply by gracefulBrown** [reply]
🤚역량 및 학습로그를 작성해야합니다.를 띄워주는 방향이 필요하지 않을 까 합니다.
- 역량이 없는 상태에서 말씀해주신 대로 알림을 띄워주고 역량 페이지로 이동 시키는게 좋겠어요!

🤚사용자의 이력 남기기가 가능한 상태일까요?
역량이 없다면 리포트를 작성 할 수 없을 것 같네요

🤚4번에서 에서 작성한 리포트를 수정한다면, '등록된 역량 현재 역량과 다릅니다.' 라는 안내가 생기는 게 맞을까요?
- 리포트 수정 기능은 없애기로 했어요!
- 대신 새로운 리포트 (이력)을 작성할 때 이전 [리포트 내용 불러오기] 기능을 추가해서 손 쉽게 변경사항을 기록할 수 있게 하려고 합니다.
- 만약 이전 리포트 작성 이후 역량이 수정되면 [리포트 내용 불러오기] 시 말씀하신대로 '등록된 역량 현재 역량과 다릅니다' 와 같은 메시지가 떠야해요! 
- 리포트 수정이라는 개념이 사라지고 무조건 리포트를 새로 작성하는 걸로 개념이 잡혔어요.
- 성장 흐름을 파악하기 위해 리포트 변경 사항을 조회

**Reply by devhyun637** [reply]
오오 그러면 '등록된 역량이 현재 역량과 다릅니다.'라는 안내만 해주고, 현재의 역량을 가져오도록 하는거군요! 감사합니다!
